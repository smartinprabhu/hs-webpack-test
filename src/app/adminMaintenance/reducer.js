const initialState = {
  createCommentInfo: {},
  orderComments: {},
  orderParts: {},
  orderCheckLists: {},
  orderTimeSheets: {},
  priorityGroupsInfo: {},
  stateGroupsInfo: {},
  stateChangeInfo: {},
  typeGroupsInfo: {},
  workordersInfo: {},
  maintenanceInfo: {},
  // workorderCount: null,
  workorderCountErr: null,
  workorderCountLoading: false,
  maintenanceCount: null,
  maintenanceCountErr: null,
  maintenanceCountLoading: false,
  workorderDetails: {},
  maintenanceExportInfo: {},
  workorderFilters: {},
  cleaningWorkorderFilters: {},
  workorderDashboard: {},
  bookingDelete: {},
  woRows: {},
  bulkOrders: {},
  workorder: {},
  workordersCount: {},
};

// eslint-disable-next-line default-param-last
function bookingMaintenanceReducer(state = initialState, action) {
  switch (action.type) {
    case 'GET_WORK_ORDERS_INFO':
      return {
        ...state,
        workordersInfo: (state.workordersInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_WORK_ORDERS_INFO_SUCCESS':
      return {
        ...state,
        workordersInfo: (state.workordersInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_WORK_ORDERS_INFO_FAILURE':
      return {
        ...state,
        workordersInfo: (state.workordersInfo, { loading: false, err: action.error, data: null }),
      };
    // case 'GET_WORK_ORDERS_COUNT':
    //   return {
    //     ...state,
    //     workorderCountLoading: true,
    //   };
    // case 'GET_WORK_ORDERS_COUNT_SUCCESS':
    //   return {
    //     ...state,
    //     workorderCount: (state.workorderCount, action.payload.data),
    //     workorderCountLoading: false,
    //   };
    // case 'GET_WORK_ORDERS_COUNT_FAILURE':
    //   return {
    //     ...state,
    //     workorderCount: (state.workorderCount, action.error),
    //     workorderCountErr: (state.workorderCountErr, action.error),
    //     workorderCountLoading: false,
    //   };
    case 'CLEAR_MAINTENANCE_INFO':
      return {
        ...state,
        maintenanceInfo: {},
      };
    case 'GET_MAINTENANCE_INFO':
      return {
        ...state,
        maintenanceInfo: (state.maintenanceInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_MAINTENANCE_INFO_SUCCESS':
      return {
        ...state,
        maintenanceInfo: (state.maintenanceInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_MAINTENANCE_INFO_FAILURE':
      return {
        ...state,
        maintenanceInfo: (state.maintenanceInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_MAINTENANCE_COUNT':
      return {
        ...state,
        maintenanceCountLoading: true,
      };
    case 'GET_MAINTENANCE_COUNT_SUCCESS':
      return {
        ...state,
        maintenanceCount: (state.maintenanceCount, action.payload),
        maintenanceCountLoading: false,
      };
    case 'GET_MAINTENANCE_COUNT_FAILURE':
      return {
        ...state,
        maintenanceCount: (state.maintenanceCount, action.error),
        maintenanceCountErr: (state.maintenanceCountErr, action.error),
        maintenanceCountLoading: false,
      };
    case 'GET_WORK_ORDERS':
      return {
        ...state,
        workorder: (state.workorder, { loading: true, data: null, err: null }),
      };
    case 'GET_WORK_ORDERS_SUCCESS':
      return {
        ...state,
        workorder: (state.workorder,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_WORK_ORDERS_FAILURE':
      return {
        ...state,
        workorder: (state.workorder, { loading: false, err: action.error, data: null }),
      };
    case 'GET_WORK_ORDERS_COUNT':
      return {
        ...state,
        workordersCount: (state.workordersCount, { loading: false, err: null, data: null }),
      };
    case 'GET_WORK_ORDERS_COUNT_SUCCESS':
      return {
        ...state,
        workordersCount: (state.workordersCount, { loading: false, err: null, data: action.payload }),
      };
    case 'GET_WORK_ORDERS_COUNT_FAILURE':
      return {
        ...state,
        workordersCount: (state.workordersCount, { loading: false, err: action.error.data, data: null }),
      };
    case 'GET_WORK_ORDERS_EXPORT_INFO':
      return {
        ...state,
        maintenanceExportInfo: (state.maintenanceExportInfo,
        { loading: true, data: null, err: null }),
      };
    case 'GET_WORK_ORDERS_EXPORT_INFO_SUCCESS':
      return {
        ...state,
        maintenanceExportInfo: (state.maintenanceExportInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_WORK_ORDERS_EXPORT_INFO_FAILURE':
      return {
        ...state,
        maintenanceExportInfo: (state.maintenanceExportInfo,
        { loading: false, err: action.error, data: null }),
      };
    case 'GET_TEAMS_GROUP_INFO':
      return {
        ...state,
        typeGroupsInfo: (state.typeGroupsInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_TEAMS_GROUP_INFO_SUCCESS':
      return {
        ...state,
        typeGroupsInfo: (state.typeGroupsInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_TEAMS_GROUP_INFO_FAILURE':
      return {
        ...state,
        typeGroupsInfo: (state.typeGroupsInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_STATES_GROUP_INFO':
      return {
        ...state,
        stateGroupsInfo: (state.stateGroupsInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_STATES_GROUP_INFO_SUCCESS':
      return {
        ...state,
        stateGroupsInfo: (state.stateGroupsInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_STATES_GROUP_INFO_FAILURE':
      return {
        ...state,
        stateGroupsInfo: (state.stateGroupsInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_PRIORITIES_GROUP_INFO':
      return {
        ...state,
        priorityGroupsInfo: (state.priorityGroupsInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_PRIORITIES_GROUP_INFO_SUCCESS':
      return {
        ...state,
        priorityGroupsInfo: (state.priorityGroupsInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_PRIORITIES_GROUP_INFO_FAILURE':
      return {
        ...state,
        priorityGroupsInfo: (state.priorityGroupsInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_ORDER_DETAILS_INFO':
      return {
        ...state,
        workorderDetails: (state.workorderDetails, { loading: true, data: null, err: null }),
      };
    case 'GET_ORDER_DETAILS_INFO_SUCCESS':
      return {
        ...state,
        workorderDetails: (state.workorderDetails, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_ORDER_DETAILS_INFO_FAILURE':
      return {
        ...state,
        workorderDetails: (state.workorderDetails, { loading: false, err: action.error, data: null }),
      };
    case 'GET_COMMENTS_INFO':
      return {
        ...state,
        orderComments: (state.orderComments, { loading: true, data: null, err: null }),
      };
    case 'GET_COMMENTS_INFO_SUCCESS':
      return {
        ...state,
        orderComments: (state.orderComments, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_COMMENTS_INFO_FAILURE':
      return {
        ...state,
        orderComments: (state.orderComments, { loading: false, err: action.error, data: null }),
      };
    case 'RESET_COMMENTS_INFO':
      return {
        ...state,
        createCommentInfo: (state.createCommentInfo, { loading: false, err: null, data: null }),
      };
    case 'CREATE_COMMENT_INFO':
      return {
        ...state,
        createCommentInfo: (state.createCommentInfo, { loading: true, data: null, err: null }),
      };
    case 'CREATE_COMMENT_INFO_SUCCESS':
      return {
        ...state,
        createCommentInfo: (state.createCommentInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'CREATE_COMMENT_INFO_FAILURE':
      return {
        ...state,
        createCommentInfo: (state.createCommentInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_STATE_CHANGE_INFO':
      return {
        ...state,
        stateChangeInfo: (state.stateChangeInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_STATE_CHANGE_INFO_SUCCESS':
      return {
        ...state,
        stateChangeInfo: (state.stateChangeInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_STATE_CHANGE_INFO_FAILURE':
      return {
        ...state,
        stateChangeInfo: (state.stateChangeInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_STATE_RESET_INFO':
      return {
        ...state,
        stateChangeInfo: (state.stateChangeInfo, { loading: false, data: null, err: null }),
      };
    case 'GET_WORKORDERS_DASHBOARD_INFO':
      return {
        ...state,
        workorderDashboard: (state.workorderDashboard, { loading: true, data: null, err: null }),
      };
    case 'GET_WORKORDERS_DASHBOARD_INFO_SUCCESS':
      return {
        ...state,
        workorderDashboard: (state.workorderDashboard, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_WORKORDERS_DASHBOARD_INFO_FAILURE':
      return {
        ...state,
        workorderDashboard: (state.workorderDashboard, { loading: false, err: action.error, data: null }),
      };
    case 'CLEAR_CLEANING_WORKORDER_FILTERS':
      return {
        ...state,
        cleaningWorkorderFilters: {},
      };
    case 'GET_PARTS_INFO':
      return {
        ...state,
        orderParts: (state.orderParts, { loading: true, data: null, err: null }),
      };
    case 'GET_PARTS_INFO_SUCCESS':
      return {
        ...state,
        orderParts: (state.orderParts, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_PARTS_INFO_FAILURE':
      return {
        ...state,
        orderParts: (state.orderParts, { loading: false, err: action.error, data: null }),
      };
    case 'GET_CHECK_LISTS_INFO':
      return {
        ...state,
        orderCheckLists: (state.orderCheckLists, { loading: true, data: null, err: null }),
      };
    case 'GET_CHECK_LISTS_INFO_SUCCESS':
      return {
        ...state,
        orderCheckLists: (state.orderCheckLists, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_CHECK_LISTS_INFO_FAILURE':
      return {
        ...state,
        orderCheckLists: (state.orderCheckLists, { loading: false, err: action.error, data: null }),
      };
    case 'GET_TIMESHEETS_INFO':
      return {
        ...state,
        orderTimeSheets: (state.orderTimeSheets, { loading: true, data: null, err: null }),
      };
    case 'GET_TIMESHEETS_INFO_SUCCESS':
      return {
        ...state,
        orderTimeSheets: (state.orderTimeSheets, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_TIMESHEETS_INFO_FAILURE':
      return {
        ...state,
        orderTimeSheets: (state.orderTimeSheets, { loading: false, err: action.error, data: null }),
      };
    case 'CLEAR_BOOKINGS_FILTER_WORKORDER':
      return {
        ...state,
        workorderFilters: {},
      };
    case 'CLEAR_BOOKINGS_WORKORDER_COUNT':
      return {
        ...state,
        maintenanceCount: {},
      };
    case 'GET_FILTER_WORKORDER':
      return {
        ...state,
        workorderFilters: (state.workorderFilters, action.payload),
      };
    case 'GET_FILTER_CLEANING_WORKORDER':
      return {
        ...state,
        cleaningWorkorderFilters: (state.cleaningWorkorderFilters, action.payload),
      };
    case 'GET_ROWS_WO':
      return {
        ...state,
        woRows: (state.woRows, action.payload),
      };
    case 'DELETE_BOOKING':
      return {
        ...state,
        bookingDelete: (state.bookingDelete, { loading: true, data: null, err: null }),
      };
    case 'DELETE_BOOKING_SUCCESS':
      return {
        ...state,
        bookingDelete: (state.bookingDelete, { loading: false, data: action.payload.data, err: null }),
      };
    case 'DELETE_BOOKING_FAILURE':
      return {
        ...state,
        bookingDelete: (state.bookingDelete, { loading: false, err: action.error.data, data: null }),
      };
    case 'SAVE_BULK_ORDERS':
      return {
        ...state,
        bulkOrders: (state.bulkOrders, { loading: true, data: null, err: null }),
      };
    case 'SAVE_BULK_ORDERS_SUCCESS':
      return {
        ...state,
        bulkOrders: (state.bulkOrders, { loading: false, data: action.payload.data, err: null }),
      };
    case 'SAVE_BULK_ORDERS_FAILURE':
      return {
        ...state,
        bulkOrders: (state.bulkOrders, { loading: false, err: action.error.data, data: null }),
      };
    case 'CLEAR_CLEANING':
      return {
        ...state,
        bulkOrders: (state.bulkOrders, { loading: false, data: null, err: null }),
      };
    case 'CLEAR_BOOKING':
      return {
        ...state,
        bookingDelete: (state.bookingDelete, { loading: false, data: null, err: null }),
      };
    default:
      return state;
  }
}

export default bookingMaintenanceReducer;
