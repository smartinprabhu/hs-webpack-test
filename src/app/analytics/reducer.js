const initialState = {
  analyticsInfo: {},
  nativeDashboard: {},
  ninjaDashboard: {},
  ninjaDashboardItem: {},
  selectedDashboard: false,
  ninjaDashboardCode: {
    loading: true,
  },
  updateLayoutInfo: {},
  sldInfo: {},
  ninjaDrillCode: {},
  ninjaDashboardDrill: {},
  updateNinjaChartItemInfo: {},
  ninjaChartInfo: {},
  dcDashboardInfo: {},
  treeDashboardInfo: {},
  ninjaDashboardExpandCode: {},
  ninjaDefaultFilters: {},
  sldRouteData: {},
  ninjaCodeCache: {},
};

function reducer(state, action) {
  if (typeof state === 'undefined') {
    return initialState;
  }
  switch (action.type) {
    case 'GET_SLD':
      return {
        ...state,
        sldInfo: (state.sldInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_SLD_SUCCESS':
      return {
        ...state,
        sldInfo: (state.sldInfo, { loading: false, err: null, data: action.payload.data }),
      };
    case 'GET_SLD_FAILURE':
      return {
        ...state,
        sldInfo: (state.sldInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_ANALYTICS_INFO':
      return {
        ...state,
        analyticsInfo: (state.analyticsInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_ANALYTICS_INFO_SUCCESS':
      return {
        ...state,
        analyticsInfo: (state.analyticsInfo, { loading: false, err: null, data: action.payload.data }),
      };
    case 'GET_ANALYTICS_INFO_FAILURE':
      return {
        ...state,
        analyticsInfo: (state.analyticsInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_NATIVE_DASHBOARD_INFO':
      return {
        ...state,
        nativeDashboard: (state.nativeDashboard, { loading: true, data: null, err: null }),
      };
    case 'GET_NATIVE_DASHBOARD_INFO_SUCCESS':
      return {
        ...state,
        nativeDashboard: (state.nativeDashboard, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_NATIVE_DASHBOARD_INFO_FAILURE':
      return {
        ...state,
        nativeDashboard: (state.nativeDashboard, { loading: false, err: action.error, data: null }),
      };
    case 'GET_NINJA_DASHBOARD_INFO':
      return {
        ...state,
        ninjaDashboard: (state.ninjaDashboard, { loading: true, data: null, err: null }),
      };
    case 'GET_NINJA_DASHBOARD_INFO_SUCCESS':
      return {
        ...state,
        ninjaDashboard: (state.ninjaDashboard, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_NINJA_DASHBOARD_TIMER_INFO_SUCCESS':
      return {
        ...state,
        ninjaDashboard: (state.ninjaDashboard, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_NINJA_DASHBOARD_INFO_FAILURE':
      return {
        ...state,
        ninjaDashboard: (state.ninjaDashboard, { loading: false, err: action.error, data: null }),
      };
    case 'GET_NINJA_DASHBOARD_ITEM_INFO':
      return {
        ...state,
        ninjaDashboardItem: (state.ninjaDashboardItem, { loading: true, data: null, err: null }),
      };
    case 'GET_NINJA_DASHBOARD_ITEM_INFO_SUCCESS':
      return {
        ...state,
        ninjaDashboardItem: (state.ninjaDashboardItem, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_NINJA_DASHBOARD_ITEM_INFO_FAILURE':
      return {
        ...state,
        ninjaDashboardItem: (state.ninjaDashboardItem, { loading: false, err: action.error, data: null }),
      };
    case 'STORE_SELECT_DASHBOARD':
      return {
        ...state,
        selectedDashboard: (state.selectedDashboard, action.payload),
      };
    case 'RESET_SELECT_DASHBOARD':
      return {
        ...state,
        selectedDashboard: (state.selectedDashboard, false),
      };
    case 'GET_ND_DETAILS_INFO':
      return {
        ...state,
        ninjaDashboardCode: (state.ninjaDashboardCode, { loading: true, data: null, err: null }),
      };
    case 'GET_ND_DETAILS_INFO_SUCCESS':
      return {
        ...state,
        ninjaDashboardCode: { loading: false, data: action.payload.data, err: null },
        ninjaCodeCache: {
          ...state.ninjaCodeCache,
          [action.id]: { data: action.payload.data, timestamp: Date.now() },
        },
      };
    case 'GET_ND_DETAILS_INFO_FAILURE':
      return {
        ...state,
        ninjaDashboardCode: (state.ninjaDashboardCode, { loading: false, err: action.error, data: null }),
      };
    case 'RESET_ND_DETAILS_INFO':
      return {
        ...state,
        ninjaDashboardCode: (state.ninjaDashboardCode, { loading: false, data: null, err: null }),
      };
    case 'UPDATE_DASHBOARD_LAYOUT_INFO':
      return {
        ...state,
        updateLayoutInfo: (state.updateLayoutInfo, { loading: true, data: null, err: null }),
      };
    case 'UPDATE_DASHBOARD_LAYOUT_INFO_SUCCESS':
      return {
        ...state,
        updateLayoutInfo: (state.updateLayoutInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'UPDATE_DASHBOARD_LAYOUT_INFO_FAILURE':
      return {
        ...state,
        updateLayoutInfo: (state.updateLayoutInfo, { loading: false, err: action.error, data: null }),
      };
    case 'RESET_UPDATE_DASHBOARD_LAYOUT_INFO':
      return {
        ...state,
        updateLayoutInfo: (state.updateLayoutInfo, { loading: false, data: null, err: null }),
      };
    case 'GET_ND_DRILL_DETAILS_INFO':
      return {
        ...state,
        ninjaDrillCode: (state.ninjaDrillCode, { loading: true, data: null, err: null }),
      };
    case 'GET_ND_DRILL_DETAILS_INFO_SUCCESS':
      return {
        ...state,
        ninjaDrillCode: (state.ninjaDrillCode, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_ND_DRILL_DETAILS_INFO_FAILURE':
      return {
        ...state,
        ninjaDrillCode: (state.ninjaDrillCode, { loading: false, err: action.error, data: null }),
      };
    case 'GET_NINJA_DASHBOARD_DRILL_INFO':
      return {
        ...state,
        ninjaDashboardDrill: (state.ninjaDashboardDrill, { loading: true, data: null, err: null }),
      };
    case 'GET_NINJA_DASHBOARD_DRILL_INFO_SUCCESS':
      return {
        ...state,
        ninjaDashboardDrill: (state.ninjaDashboardDrill, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_NINJA_DASHBOARD_DRILL_INFO_FAILURE':
      return {
        ...state,
        ninjaDashboardDrill: (state.ninjaDashboardDrill, { loading: false, err: action.error, data: null }),
      };
    case 'GET_NINJA_DASHBOARD_TIMER_DRILL_INFO_SUCCESS':
      return {
        ...state,
        ninjaDashboardDrill: (state.ninjaDashboardDrill, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_CHART_ITEM_DETAILS_INFO':
      return {
        ...state,
        ninjaChartInfo: (state.ninjaChartInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_CHART_ITEM_DETAILS_INFO_SUCCESS':
      return {
        ...state,
        ninjaChartInfo: (state.ninjaChartInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_CHART_ITEM_DETAILS_INFO_FAILURE':
      return {
        ...state,
        ninjaChartInfo: (state.ninjaChartInfo, { loading: false, err: action.error, data: null }),
      };
    case 'UPDATE_DASHBOARD_ITEM_INFO':
      return {
        ...state,
        updateNinjaChartItemInfo: (state.updateNinjaChartItemInfo, { loading: true, data: null, err: null }),
      };
    case 'UPDATE_DASHBOARD_ITEM_INFO_SUCCESS':
      return {
        ...state,
        updateNinjaChartItemInfo: (state.updateNinjaChartItemInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'UPDATE_DASHBOARD_ITEM_INFO_FAILURE':
      return {
        ...state,
        updateNinjaChartItemInfo: (state.updateNinjaChartItemInfo, { loading: false, err: action.error, data: null }),
      };
    case 'RESET_UPDATE_DASHBOARD_ITEM_INFO':
      return {
        ...state,
        updateNinjaChartItemInfo: (state.updateNinjaChartItemInfo, { loading: false, err: null, data: null }),
      };
    case 'GET_DC_DASHBOARD_INFO':
      return {
        ...state,
        dcDashboardInfo: (state.dcDashboardInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_DC_DASHBOARD_INFO_SUCCESS':
      return {
        ...state,
        dcDashboardInfo: (state.dcDashboardInfo, { loading: false, err: null, data: action.payload.data }),
      };
    case 'GET_DC_DASHBOARD_INFO_FAILURE':
      return {
        ...state,
        dcDashboardInfo: (state.dcDashboardInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_TREE_DASHBOARD_DETAILS_INFO':
      return {
        ...state,
        treeDashboardInfo: (state.treeDashboardInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_TREE_DASHBOARD_DETAILS_INFO_SUCCESS':
      return {
        ...state,
        treeDashboardInfo: (state.treeDashboardInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_TREE_DASHBOARD_DETAILS_INFO_FAILURE':
      return {
        ...state,
        treeDashboardInfo: (state.treeDashboardInfo, { loading: false, err: action.error, data: null }),
      };
    case 'RESET_TREE_DASHBOARD_DETAILS_INFO':
      return {
        ...state,
        treeDashboardInfo: (state.treeDashboardInfo, { loading: false, err: null, data: null }),
      };
    case 'GET_ND_DETAILS_EXPAND_INFO':
      return {
        ...state,
        ninjaDashboardExpandCode: (state.ninjaDashboardExpandCode, { loading: true, data: null, err: null }),
      };
    case 'GET_ND_DETAILS_EXPAND_INFO_SUCCESS':
      return {
        ...state,
        ninjaDashboardExpandCode: (state.ninjaDashboardExpandCode, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_ND_DETAILS_EXPAND_INFO_FAILURE':
      return {
        ...state,
        ninjaDashboardExpandCode: (state.ninjaDashboardExpandCode, { loading: false, err: action.error, data: null }),
      };
    case 'RESET_ND_DETAILS_EXPAND_INFO':
      return {
        ...state,
        ninjaDashboardExpandCode: (state.ninjaDashboardExpandCode, { loading: false, data: null, err: null }),
      };
    case 'GET_DEFAULT_FILTERS_INFO':
      return {
        ...state,
        ninjaDefaultFilters: (state.ninjaDefaultFilters, { loading: true, data: null, err: null }),
      };
    case 'GET_DEFAULT_FILTERS_INFO_SUCCESS':
      return {
        ...state,
        ninjaDefaultFilters: (state.ninjaDefaultFilters, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_DEFAULT_FILTERS_INFO_FAILURE':
      return {
        ...state,
        ninjaDefaultFilters: (state.ninjaDefaultFilters, { loading: false, err: action.error, data: null }),
      };
    case 'RESET_DEFAULT_FILTERS_INFO':
      return {
        ...state,
        ninjaDefaultFilters: (state.ninjaDefaultFilters, { loading: false, data: null, err: null }),
      };
    case 'RESET_NINJA_DASHBOARD_INFO':
      return {
        ...state,
        ninjaDashboard: (state.ninjaDashboard, { loading: false, data: null, err: null }),
      };
    case 'STORE_SLD_DATA':
      return {
        ...state,
        sldRouteData: (state.sldRouteData, action.payload),
      };
    case 'GET_ND_DETAILS_FROM_CACHE':
      return {
        ...state,
        ninjaDashboardCode: { loading: false, data: action.payload.data, err: null },
      };
    default:
      return state;
  }
}

export default reducer;
