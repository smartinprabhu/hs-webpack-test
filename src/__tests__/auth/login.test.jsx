import React from 'react';
import { useSelector } from 'react-redux';
import {
  render, fireEvent, screen,
} from '@testing-library/react';
import fetchMock from 'fetch-mock';

import '@testing-library/jest-dom/extend-expect';

import Login from '../../app/auth/login';
import * as actions from '../../app/auth/authActions';
import reducer from '../../app/auth/authReducer';

jest.mock('react-redux', () => ({
  useSelector: jest.fn((fn) => fn()),
  useDispatch: () => jest.fn(),
}));

describe('login testing', () => {
  afterEach(() => {
    fetchMock.restore();
  });

  describe('login snapshots', () => {
    it('snapshot testing', () => {
      useSelector.mockImplementation((fn) => fn({
        auth: {
          authInfo: null,
          authError: null,
        },
      }));
      const { asFragment } = render(
        <Login />,
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('login actions, reducers', () => {
    const initialState = {
      authError: null,
      authInfo: null,
    };

    it('should return the initial state', () => {
      expect(reducer(undefined, {})).toEqual(initialState);
    });

    const loginRes = {
      access_token: 'fpC8oaGlwknbFbJpHRSmijBvbp53tJ',
      expires_in: 3600,
      token_type: 'Bearer',
      scope: '',
      refresh_token: 'F4Lub38xxOBubAlxjAqQMPcLl68BH6',
    };

    it('it should return the LOGIN_SUCCESS', () => {
      expect(reducer(initialState, {
        type: actions.LOGIN_SUCCESS,
        payload: loginRes,
      })).toEqual({
        authInfo: loginRes,
        authError: null,
      });
    });

    it('it should return the LOGIN_FAILURE', () => {
      expect(reducer(initialState, {
        type: actions.LOGIN_FAILURE,
        error: 'Login Failed',
      })).toEqual({
        authInfo: null,
        authError: 'Login Failed',
      });
    });

    it('it should return the LOGOUT_SUCCESS', () => {
      expect(reducer(initialState, {
        type: actions.LOGOUT_SUCCESS,
      })).toEqual({
        authInfo: null,
        authError: null,
      });
    });

    it('it should return the LOGOUT_FAILURE', () => {
      expect(reducer(initialState, {
        type: actions.LOGOUT_FAILURE,
        error: 'Logout Failed',
      })).toEqual({
        authInfo: null,
        authError: 'Logout Failed',
      });
    });

    it('should handle login failure', () => {
      const data = 'Login Failed';
      const expectedAction = {
        type: actions.LOGIN_FAILURE,
        payload: data,
      };
      expect(actions.loginFailure(data)).toEqual(expectedAction);
    });

    it('should handle request for login success', () => {
      const data = 'success';
      const expectedAction = {
        type: actions.LOGIN_SUCCESS,
        payload: data,
      };
      expect(actions.loginSuccess(data)).toEqual(expectedAction);
    });

    it('should handle logout failure', () => {
      const data = 'Logout Failed';
      const expectedAction = {
        type: actions.LOGOUT_FAILURE,
        payload: data,
      };
      expect(actions.logoutFailure(data)).toEqual(expectedAction);
    });

    it('should handle request for logout success', () => {
      const expectedAction = {
        type: actions.LOGOUT_SUCCESS,
      };
      expect(actions.logoutSuccess()).toEqual(expectedAction);
    });
  });

  const userLoginMock = {
    username: 'technopark@helixsense.com',
    password: 'p',
  };

  it('should test Login', () => {
    const handleSubmit = jest.fn();
    const { container, getByText } = render(<Login handleSubmit={handleSubmit} />);
    const username = screen.getByLabelText('Username', { selector: 'input' });
    const password = screen.getByLabelText('Password', { selector: 'input' });
    fireEvent.change(username, {
      target: {
        value: userLoginMock.username,
      },
    });

    fireEvent.change(password, {
      target: {
        value: userLoginMock.password,
      },
    });

    expect(username).toHaveAttribute('id', 'exampleUsername');
    expect(password).toHaveAttribute('id', 'examplePassword');
    fireEvent.click(getByText('LOGIN'));
    expect(container).toBeInTheDocument();
  });

  it('display error message on login faliure', () => {
    useSelector.mockImplementation((fn) => fn({
      auth: {
        authInfo: null,
        authError: 'Logout Failed',
      },
    }));

    const { container } = render(
      <Login />,
    );
    expect(screen.getByRole('alert')).toHaveTextContent('Logout Failed');
    expect(container).toBeInTheDocument();
  });
});
