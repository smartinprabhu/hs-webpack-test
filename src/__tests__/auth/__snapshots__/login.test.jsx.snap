// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login testing login snapshots snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="login"
  >
    <div
      class="m-0 row"
    >
      <div
        class="bg-white p-4 mt-7 opacity-9 col-sm-4 order-sm-1 offset-sm-7 col-md-4 order-md-1 offset-md-7 col-lg-4 order-lg-1 offset-lg-7"
      >
        <div
          class="m-0 row"
        >
          <div
            class="pl-0 col-sm-9 col-md-9 col-lg-9"
          >
            <h2>
              Welcome back!
            </h2>
            <span
              class="light-text"
            >
              Please sign in to your account.
            </span>
          </div>
          <div
            class="col-sm-3 col-md-3 col-lg-3"
          >
            <img
              alt="Helixsense Portal"
              class="float-right"
              src="test-file-stub"
              width="80"
            />
          </div>
        </div>
        <form
          class="mt-3"
          novalidate=""
        >
          <div
            class="form-group"
          >
            <label
              class=""
              for="exampleUsername"
            >
              Username
            </label>
            <input
              class="border-radius-50px form-control"
              id="username"
              name="username"
              placeholder="Username"
              type="username"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class=""
              for="examplePassword"
            >
              Password
            </label>
            <input
              class="border-radius-50px form-control"
              id="password"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <button
            class="roundCorners p-login btn btn-secondary"
            type="submit"
          >
            <span
              class="ml-2"
            />
            LOGIN
          </button>
          <button
            class="roundCorners p-login float-right btn btn-secondary"
            type="button"
          >
            REGISTER
          </button>
          <div
            class="mt-2"
          >
            <button
              class="cursor-pointer btn btn-link"
              type="button"
            >
              Forgot Password?
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
